src/components/timer/Pie.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+export default class Pie extends React.Component {
+  constructor(props) {
+     super(props);
+    this.state = {style: {}, x: 400, y: 200, slice: (360/this.props.time), totalSlices: this.props.time};
+    this.draw = this.draw.bind(this);
+  }
+
+  draw() {
+    let radians = ( this.state.totalSlices - this.props.time + 1 )*this.state.slice*(Math.PI / 180);
+    let radius = 200, cx = 200, cy = 200;
+    let x = cx + radius*Math.cos(radians);
+    let y = cy + radius*Math.sin(radians);
+    this.setState( {x: x,y: y} );
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    if( this.props.time != nextProps.time )
+      return true;
+    else
+      return false;
+    if( this.state.x != nextState.x ) {
+      return true;
+    }
+  }
+
+  componentDidUpdate() {
+    this.draw();
+  }
+
+  render() {    
+    //todo: supress warning on anti-pattern
+    this.draw();
```

Comment:
The draw method calls setState. You shouldn't call setState in the render method. There really isn't a need to store state for this component. You can just use local variables.

Reply:
Confirming and updated. Conformed to local variables. No longer following an _anti-pattern_.